tests/csm/rest/test_capacity_usage.py

```diff
@@ -1968,3 +2018,914 @@ def test_34723(self):
+        self.log.info("-------------------------Step 1 Starts-------------------------")
+        self.log.info(
+            "[Start] Fetch degraded capacity on Consul with 0 Node failure")
+        resp = self.csm_obj.get_capacity_consul()
+        cap_df = self.csm_obj.get_dataframe_failure_recovery(self.num_worker)
+        test_cfg = self.csm_conf["test_39924"]
+        total_written = resp["healthy"]
+        cap_df.loc["No failure"]["consul_healthy"] = resp["healthy"]
+        cap_df.loc["No failure"]["consul_degraded"] = resp["degraded"]
+        cap_df.loc["No failure"]["consul_critical"] = resp["critical"]
+        cap_df.loc["No failure"]["consul_damaged"] = resp["damaged"]
+        assert self.csm_obj.verify_degraded_capacity(resp, healthy=None, degraded=0,
+                                                     critical=0, damaged=0,
+                                                     err_margin=test_cfg["err_margin"])
+        self.log.info(
+            "[End] Fetch degraded capacity on Consul with 0 Node failure")
+        self.log.info(
+            "[Start] Fetch degraded capacity on HCTL with 0 Node failure")
+        resp = self.hlth_master.hctl_status_json()["bytecount"]
+        cap_df.loc["No failure"]["hctl_healthy"] = resp["healthy"]
+        cap_df.loc["No failure"]["hctl_degraded"] = resp["degraded"]
+        cap_df.loc["No failure"]["hctl_critical"] = resp["critical"]
+        cap_df.loc["No failure"]["hctl_damaged"] = resp["damaged"]
+        assert self.csm_obj.verify_degraded_capacity(
+            resp, healthy=None, degraded=0, critical=0, damaged=0,
+            err_margin=test_cfg["err_margin"],
+            total=total_written)
+        self.log.info(
+            "[End] Fetch degraded capacity on HCTL with 0 Node failure")
+        self.log.info("[Start] Fetch degraded capacity on CSM")
+        resp = self.csm_obj.get_degraded_capacity()
+        assert resp.status_code == HTTPStatus.OK, "Status code check failed."
+        resp = resp.json()["bytecount"]
+        cap_df.loc["No failure"]["csm_healthy"] = resp["healthy"]
+        cap_df.loc["No failure"]["csm_degraded"] = resp["degraded"]
+        cap_df.loc["No failure"]["csm_critical"] = resp["critical"]
+        cap_df.loc["No failure"]["csm_damaged"] = resp["damaged"]
+        healthy_count = resp["healthy"]
+        assert self.csm_obj.verify_degraded_capacity(
+            resp, healthy=None, degraded=0, critical=0, damaged=0,
+            err_margin=test_cfg["err_margin"],
+            total=total_written)
+        self.log.info(
+            "[End] Fetch degraded capacity on CSM with 0 Node failure")
+        self.log.info("-----------------------Step 1 Ends--------------------------")
+        self.log.info("Get the value of K for the given cluster.")
+        resp = self.ha_obj.get_config_value(self.master)
+        if resp[0]:
+            self.kvalue = int(resp[1]['cluster']['storage_set'][0]['durability']['sns']['parity'])
```

Comment:
Can u move this to setup_class as it will be required by other tests also.

Reply:
okay